Minio/DataModel/ObjectOperationsArgs.cs

```diff
@@ -853,4 +855,357 @@ public override RestRequest BuildRequest(RestRequest request)
+            {
+                parts.Add(new XElement("Part",
+                                       new XElement("PartNumber", i),
+                                       new XElement("ETag", this.ETags[i])));
+            }
+
+            var completeMultipartUploadXml = new XElement("CompleteMultipartUpload", parts);
+            request.XmlSerializer = new RestSharp.Serializers.DotNetXmlSerializer();
+            request.XmlSerializer.Namespace = "http://s3.amazonaws.com/doc/2006-03-01/";
+            request.XmlSerializer.ContentType = "application/xml";
+            string body = utils.MarshalXML(completeMultipartUploadXml, "http://s3.amazonaws.com/doc/2006-03-01/");
+            request.AddParameter("text/xml", body, ParameterType.RequestBody);
+            return request;
+        }
+
+        public override void Validate()
+        {
+            base.Validate();
+            if (string.IsNullOrEmpty(this.UploadId) || string.IsNullOrWhiteSpace(this.UploadId))
+            {
+                throw new ArgumentNullException(nameof(UploadId) + " not assigned for CompleteMultipartUpload operation.");
+            }
+        }
+    }
+    public class PutObjectArgs : ObjectWriteArgs<PutObjectArgs>
+    {
+        internal string UploadId { get; private set; }
+        internal int PartNumber { get; set; }
+        internal string FileName { get; set; }
+        internal long ObjectSize { get; set; }
+        internal Stream ObjectStreamData { get; set; }
+
+        public PutObjectArgs()
+        {
+            this.RequestMethod = Method.PUT;
+        }
+
+        public PutObjectArgs(PutObjectPartArgs args)
+        {
+            this.RequestMethod = Method.PUT;
+            this.BucketName = args.BucketName;
+            this.ContentType = args.ContentType ?? "application/octet-stream";
+            this.FileName = args.UploadId;
+            this.HeaderMap = args.HeaderMap;
+            this.ObjectName = args.ObjectName;
+            this.ObjectSize = args.ObjectSize;
+            this.PartNumber = args.PartNumber;
+            this.SSE = args.SSE;
+            this.SSEHeaders = args.SSEHeaders;
+            this.UploadId = args.UploadId;
+        }
+
+        public override void Validate()
+        {
+            base.Validate();
+            if (this.RequestBody == null && this.ObjectStreamData == null && string.IsNullOrEmpty(this.FileName))
+            {
+                throw new ArgumentNullException("Invalid input. " + nameof(RequestBody) + ", " + nameof(FileName) + " and " + nameof(ObjectStreamData) + " cannot be empty.");
+            }
+            if (this.PartNumber < 0 )
+            {
+                throw new ArgumentOutOfRangeException(nameof(PartNumber), this.PartNumber, "Invalid Part number value. Cannot be less than 0");
+            }
+            if (!string.IsNullOrEmpty(this.FileName))
+            {
+                utils.ValidateFile(this.FileName);
+            }
+            this.Populate();
```

Comment:
since you are combining multiple overloaded calls into one call, you will need to guard against user entering conflicting data - such as providing filename and a stream

Reply:
Yes, Validate() call is updated.